src/v2/ion/ViewClassNamesFactory.js

```diff
@@ -0,0 +1,88 @@
+/* eslint complexity: [2, 25] */
+import {FORMS} from './RemediationConstants';
+
+const FORMNAME_CLASSNAME_MAPPINGS = {
+  [FORMS.IDENTIFY]: {
+    'identify': ['identify', 'primary-auth'],
+    'password': ['identify', 'primary-auth']
+  },
+  [FORMS.ENROLL_PROFILE]: {
+    'enroll-profile': ['enroll-profile', 'registration'],
+  },
+  [FORMS.CHALLENGE_AUTHENTICATOR]: {
+    email: ['challenge-authenticator-email', 'mfa-verify-passcode'],
+    password: ['challenge-authenticator-password', 'mfa-verify-password'],
+    sms: ['challenge-authenticator-voice', 'mfa-verify-passcode'],
+    voice: ['challenge-authenticator-voice', 'mfa-verify-passcode'],
+    'security_question': ['challenge-authenticator-security_question', 'mfa-verify-question']
+  },
+  [FORMS.ENROLL_AUTHENTICATOR]: {
+    email: ['enroll-authenticator-email', 'enroll-email'],
+    password: ['enroll-authenticator-password', 'enroll-password'],
+    sms: ['enroll-authenticator-sms', 'enroll-sms'],
+    voice: ['challenge-authenticator-voice', 'mfa-verify-passcode'],
+    'security_question': ['enroll-authenticator-security_question', 'enroll-question'],
+  },
+  [FORMS.AUTHENTICATOR_ENROLLMENT_DATA]: {
+    voice: ['authenticator-enrollment-data'],
+    sms: ['authenticator-enrollment-data']
+  },
+  [FORMS.AUTHENTICATOR_VERIFICATION_DATA]: {
+    voice: ['authenticator-verification-data'],
+    sms: ['authenticator-verification-data']
+  },
+  [FORMS.SELECT_AUTHENTICATOR_ENROLL]: {
+    'select-authenticator-enroll': ['select-authenticator-enroll', 'enroll-choices']
+  },
+  [FORMS.SELECT_AUTHENTICATOR_AUTHENTICATE]: {
+    'password': ['select-authenticator-authenticate']
+  },
+  [FORMS.REENROLL_AUTHENTICATOR]: {
+    'password':['reenroll-authenticator', 'password-expired']
+  },
+  [FORMS.REENROLL_AUTHENTICATOR_WARNING]: {
+    'password':['reenroll-authenticator_warning']
+  },
+  [FORMS.RESET_AUTHENTICATOR]: {
+    'password':['reset-authenticator', 'forgot-password']
+  },
+};
+
+const getClassNameMapping = (appState) => {
+  const formName = appState.get('currentFormName');
+  const authenticator = appState.get('currentAuthenticatorEnrollment') || appState.get('currentAuthenticator');
+  let authenticatorType = authenticator && authenticator.type;
+  const methodType = authenticator && authenticator.methods && authenticator.methods[0].type;
+  const isPasswordRecoveryFlow = appState.get('recoveryAuthenticator') &&
```

Comment:
replaced with `isPasswordRecovery` ?

Reply:
Referencing the derived property from appState actually exposing a bug we have where the derived properties are not getting updated correctly when the ION response changes. When we clear the appState, it still keeps the values for the derived properties intact causing randomness.